service/history/workflow/retry.go

```diff
@@ -241,13 +241,14 @@ func SetupNewWorkflowForRetryOrCron(
 	}
 
 	req := &historyservice.StartWorkflowExecutionRequest{
-		NamespaceId:              newMutableState.GetNamespaceEntry().ID().String(),
-		StartRequest:             createRequest,
-		ParentExecutionInfo:      parentInfo,
-		LastCompletionResult:     lastCompletionResult,
-		ContinuedFailure:         failure,
-		ContinueAsNewInitiator:   initiator,
-		FirstWorkflowTaskBackoff: timestamp.DurationPtr(backoffInterval),
+		NamespaceId:            newMutableState.GetNamespaceEntry().ID().String(),
+		StartRequest:           createRequest,
+		ParentExecutionInfo:    parentInfo,
+		LastCompletionResult:   lastCompletionResult,
+		ContinuedFailure:       failure,
+		ContinueAsNewInitiator: initiator,
+		// enforce minimal interval between runs to prevent tight loop continue as new spin.
+		FirstWorkflowTaskBackoff: previousMutableState.ContinueAsNewMinBackoff(&backoffInterval),
```

Comment:
Could this be behavior change for existing cron and retry?

Reply:
It won't be for cron because cron cannot schedule faster than every 1s. For retry, I think it is ok to enforce 1s minimal interval.